gpii/node_modules/flowManager/src/PrefsServerDataSource.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * GPII Preferences Server Data Source
+ *
+ * Copyright 2018 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+// "gpii.flowManager.prefsServerDataSource" is used by the cloud based flow manager to communicate with the
+// preferences server.
+fluid.defaults("gpii.flowManager.prefsServerDataSource", {
```

Comment:
I'm a bit puzzled where all this impl came from since I don't see a big negative diff elsewhere in the pull?

Reply:
`prefsServerDataSource` is to replace [the previous `preferencesDataSource`](https://github.com/GPII/universal/pull/699/files#diff-5849dc7399970f8a5a589c7764012d2fL31). The old `preferencesDataSource` only needs to read/write preferences from /preferences endpoint on the prefs server. As now the cloud based flow manager also needs to ping /ready endpoint, I added this new data source for handling requests to multiple endpoints on the prefs server.